builder/steps/gen-dockerfile/contents/src/detect_setup.ts

```diff
@@ -161,10 +162,12 @@ export async function detectSetup(
         'the "server.js" file were found.');
   }
 
-  return {
+  // extend filters undefined properties.
+  const setup = extend({}, {
```

Comment:
I agree.  I don't think there was any particular reason why I didn't just return the result of `extend` directly.